src/Services/PaymentService.php

```diff
@@ -123,6 +124,10 @@ public function authorize(
             $authorization->setCard3ds($card3ds);
         }
         $payment->setAuthorization($authorization)->setCustomer($customer)->setMetadata($metadata)->setBasket($basket);
+
+        if ($recurrenceType !== null) {
+            $authorization->setRecurrenceType($recurrenceType);
+        }
```

Comment:
Didn't you want to move the check to the setRecurrenceType method?

Reply:
I decided differently after taking another look, since it would change things more then first expected.

Reply:
You should really move it.
It is the maintainability that will suffer, and refactoring is a necessary step that will show you how you can avoid things like that in the future.
This is a redundancy and that is a code smell.

Reply:
It is real:
https://medium.com/@matryer/broken-windows-theory-why-code-quality-and-simplistic-design-are-non-negotiable-e37f8ce23dab

Reply:
After inspecting the code, it does not seem to be feasible to change the setter function to allow null values.